polygon/heimdall/reader.go

```diff
@@ -0,0 +1,57 @@
+package heimdall
+
+import (
+	"context"
+
+	libcommon "github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/log/v3"
+	"github.com/erigontech/erigon/polygon/bor/valset"
+)
+
+type Reader struct {
+	logger                    log.Logger
+	store                     ServiceStore
+	spanBlockProducersTracker *spanBlockProducersTracker
+}
+
+// AssembleReader creates and opens the MDBX store. For use cases where the store is only being read from. Must call Close.
+func AssembleReader(ctx context.Context, calculateSprintNumber CalculateSprintNumberType, dataDir string, tmpDir string, logger log.Logger) (*Reader, error) {
```

Comment:
I would suggest `CalculateSprintNumberFunction` to make it clear this is a function.

Reply:
Renamed to [CalculateSprintNumberFunc](https://github.com/erigontech/erigon/pull/11924/commits/994d100f707fba8fb53652b3344bea1510b0b197)